### ISSUE TITLE
Proposal: refactor validtors of cobra commond args

### ISSUE THREAD

**Issue by aFlyBird0** [devstream-io/devstream#478]
## Description

To make validators more elegant:
1. I want to separate args validators and important business code.
2. I want to reduce duplicate code of validators.

## Describe the Proposal

1. For code separate
take `show` cmd for example:
```Go
func showCMDFunc(cmd *cobra.Command, args []string) {
	if err := validateShowArgs(args); err != nil {
		log.Fatal(err)
	}

	// There is important business code
}
```
I will delete code like `if err:= validate...Show(args); err!=nil{log.Fatal(err)` in all commonds.

**New code will be like:**  
```Go
var showCMD = &cobra.Command{
	Use:   "...",
	Short: "...",
	Long: `...`,
	Run: WithValidators(showCMDFunc, validator1, validator2),
}

func showCMDFunc(cmd *cobra.Command, args []string) {
	// Only important business code and following business errors
}
```

2. For duplicate code
There is duplicate code like:
```
if len(args) != 1 {
		return fmt.Errorf("illegal args count (expect 1, got %d)", len(args))
	}
```
I will define some common validators to program easier and keep  error messages consistent.

However, I'am not sure where to put code about validators interface, maybe `/cmd/devstream/validator/validator.go` ? Or just in  `/cmd/devstream/validator.go` to hide pkg names. Can you give me some advice? @daniel-hutao

**Comment by daniel-hutao** [comment]
`/cmd/devstream/validator/validator.go` looks better. Because maybe we'll add many validators in some separated go file under the `validator dir`.

**Comment by IronCore864** [comment]
Hi @aFlyBird0, thanks for your enthusiastic contribution!

I have read your PR already; sorry for the late reply. I'll reply directly to the issue instead of in the PR.

We follow the [standard go project layout](https://github.com/golang-standards/project-layout), and according to those conventions, the [cmd](https://docs.devstream.io/en/latest/contributing/project-layout/#cmd) folder shouldn't contain a lot of code.

If you think the code can be imported and used in other projects, then it should live in the `/pkg` directory. If the code is not reusable or if you don't want others to reuse it, put that code in the `/internal` directory.

In this case, I think the [validator.go](https://github.com/devstream-io/devstream/pull/479/files#diff-58d5d1452a41ec04bc76306c7510830bec83a45e3fc5480aaaf36ebc4e5ebf21) file is better to be put under `/internal/pkg`. Maybe even a new subfolder there? 

Feel free to have a discussion here or in our WeChat user group or slack channel :)

Thanks again @aFlyBird0

**Comment by daniel-hutao** [comment]
![](https://github.com/devstream-io/devstream/raw/main/docs/images/wechat-group-qr-code.jpg)

@aFlyBird0

**Comment by daniel-hutao** [comment]
@IronCore864 @aFlyBird0 

The validator is a kind of entry logic. So it is more suitable under `cmd`. For example, k8s does this:

![RUq0Rl59Vl](https://user-images.githubusercontent.com/18692628/168193422-0926ead5-686e-4c20-8ca8-d516acf12788.jpg)

**Comment by aFlyBird0** [comment]
Thank you both for your advice. @daniel-hutao @IronCore864 I have finished coding

**Comment by daniel-hutao** [comment]
closed by #479
